### DISCUSSION TITLE
Tool that helps you doing the distortion calibration

### DISCUSSION THREAD

**Discussion by uchrisu** [lensfun/lensfun#1610] [General]
Since the TCA and vignetting is automated and the task of distortion calibration using Hugin takes a long time, I wrote a little python script some time ago that helps me doing this job more efficiently. 
Now I have published it and uploaded a little video explaining the tool. 

You can find the tool here: https://github.com/uchrisu/lf_autodistortion

The introduction video you can find here: https://youtu.be/BhJ_lWgMc64

The Tool is not perfect, so I will improve it in the future when I find some time for it. 

Maybe it can already help the calibrators to save a lot of time.

**Comment by sarunasb** [comment]
Your tool is very useful already, especially on good calibration images.

Would it be possible to make the detected line more visible? Thicker and/or some outlandish color?

When lines in image are not contrasty enough, there might be quite some jitter in the detected line. I don't know about the fitting algorithm used. Is this high frequency filtered out somehow?

**Comment by uchrisu** [comment]
Thanks for your suggestions. 
The lines should be visible when you zoom in. Since I use the simple High GUI interface of opencv the presentation is not perfect and might depend on your opencv package and configuration. I tried to improve it a little bit (i updated the repository). 

It also seemed as if I uploaded an old version for of the line-finding algorithm. Maybe you will get better results yet. 

Some jitter should be averaged out, since the whole line has to be fitted to a third order polynomial (model used by lensfun).

**Reply by sarunasb** [reply]
Lines are better visible now, thanks!
The tool is very helpful.
